.github/workflows/deploy.yml

```diff
@@ -12,13 +12,15 @@ jobs:
     if: startsWith(github.event.ref, 'refs/tags')
 
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v4
       - name: Set up Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v5
         with:
-          python-version: "3.10"
+          python-version: "3.12"
```

Comment:
We can consider using `3.x` which means "latest stable version".

Reply:
Done